docs/development/development.md

```diff
@@ -0,0 +1,76 @@
+# Development FAQ
+
+### **What are the points of the library used by furyctl that are not automatically generated and what's their purpose?**
+
+<details>
+  <summary>Answer</summary>
+
+The files used by `furyctl`, in addition to the schema generated in `pkg/apis`, are:
+
+- **Defaults for the distribution**: These are contained in the `defaults` folder. These files define the default values and settings that are used for the configuration, ensuring that the distribution behaves in a predictable manner when no explicit configurations are provided.
+
+- **Templates for the distribution**: These are found in the `templates` folder and include specific templates like Terraform for `EKSCluster`. These templates serve as blueprints for configuring and deploying components (like cloud resources), enabling customizations for different environments.
+
+- **Rules for `furyctl.yaml` changes**: The rules determine which changes are permitted in the `furyctl.yaml` file after initial deployment. These rules are critical for ensuring that modifications to the configuration are safe and consistent with the intended deployment workflow, particularly when managing state changes across environments.
+</details>
+
+---
+
+### **How do the rules for immutable fields/migrations work and where are they evaluated in furyctl's code?**
+
+<details>
+  <summary>Answer</summary>
+
+The rules for immutable fields and migrations are evaluated within the core logic of the `furyctl` tool. These rules are defined in the configuration files and enforced by the tool to ensure consistency and prevent misconfigurations.
+
+These rules serve as safety mechanisms during module changes (e.g., switching from the Loki logging system to OpenSearch). Some changes are allowed, while others are not. For example, fields marked as `immutable` will return an error if an attempt is made to change them.
+
+The rules are configured in the `rules` folder, and the commands/scripts executed for each rule are found in `templates/distribution/scripts/pre-apply.sh.tpl`.
+
+Additionally, **Reducers** are special fields rendered inside the template engine that indicate whether a particular feature or module of the distribution has changed. The `.to` and `.from` strings indicate these changes precisely.
```

Comment:
Let's add some info about the key of the value that can be used in the template engine, to know when something changes